content/blog/willing-to-build-up-your-own-workshops-on-demand-infrastructure.md

```diff
@@ -0,0 +1,135 @@
+
+![](/img/howto-wod-4.png)
+
+### T﻿he Register Phase:
+
+T﻿he participant starts by browsing a frontend web server that presents the catalog of available workshops. He then select one of them and registers for it by entering his Email address, first and last names as well as his company name. Finally he accepts the terms and conditions and hit the register button. As he is clicking the register button, the frontend server will perform a series of actions:
+
+1.Assign a student from the dedicated workshop range to the participant. Every workshop has a dedicated range of student assigned to it.
+
+H﻿ere is a screenshot of the workshop table present in the frontend Database Server showing API101 workshops details.
+
+![Workshops Table from Frontend DB server](/img/howto-wod-2.png "Workshops Table from Frontend DB server")
+
+* Frederic Passeron gets assigned a studentid "student397" for workshop "API101".
+
+![Customers Table from Frontend DB server](/img/howto-wod-3.png "Customers Table from Frontend DB server")
+
+H﻿ere are the details of the participant info when registered to a given workshop.
+
+2﻿.A first email is sent to participant from the frontend server welcoming him to the workshop and informing him the deployment is ongoing and that a second email show be recieved shortly providing him with the necessary infos to logon  to the workshop environment.
+
+3﻿.The frontend server at the same time sends the necessary orders through a procmail API to the backend server. The mail sent to the backend server contains the following details :
+
+* Action Type ( CREATE, CLEANUP, RESET)
+* W﻿orkshop ID
+* S﻿tudent ID
+
+4﻿.The Backend Server recieves the order and processes it by  parsing the email recieved using the procmail API. the procmail API automates the managementof the workshops.
+
+Like any API, it uses verbs to perform tasks.
+
+* CREATE to deploy a workshop
+* C﻿LEANUP to delete a workshop
+* R﻿ESET to reset associated workshop's resource
+
+T﻿his means:
+
+* P﻿reparing any infrastructure that might be requiered for the workshop (Virtual Appliance, Virtual Machine, Docker Container, LDAP config, etc..)
+* G﻿enerate a random Password for the allocated student
+* D﻿eploy the workshop content on the jupyterhub server in the dedicated student home directory (Notebooks files necessary for the workshop)
+* Send back through API Calls to the frontend server the confirmation of the deployment of the workshop along with the student details (Password)
+
+5﻿.The frontend server get its different tables updated:
+
+* T﻿he customer tables shows an active status for the participant row. The password field has been updated. 
+* T﻿he Workshop table gets also updated. The capacity field decrement the number of available seats. 
+* The student tables gets updated as well by setting the allocated student to active.
+
+6﻿.The frontend server sends the second email to the particpant providing him with the details to connect to the workshop environment.
```

Comment:
Rewrite to:
"6. The frontend server sends the second email to each participant providing them with the details to connect to the workshop environment."

[There is a reason why I'm recommending you stay away from using just a singular "he" as the subject in each of these instances.]